plugins/osdn/podinfoserver.go

```diff
@@ -0,0 +1,138 @@
+package osdn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/gorilla/mux"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+)
+
+type PodInfoServer struct {
+	http.Server
+
+	registry *Registry
+	vnidMap  *VNIDMap
+	nodeName string
+
+	clusterNetwork string
+	nodeNetwork    string
+	nodeGateway    string
+	mtu            uint
+}
+
+func NewPodInfoServer(r *Registry, v *VNIDMap, nodeName string) *PodInfoServer {
+	router := mux.NewRouter()
+ 
+	s := &PodInfoServer{
+		registry: r,
+		vnidMap:  v,
+		nodeName: nodeName,
+		Server:   http.Server{
+			Handler: router,
+		},
+	}
+
+	router.NotFoundHandler = http.HandlerFunc(http.NotFound)
+	router.Methods("GET").Path("/network").HandlerFunc(s.getNetwork)
```

Comment:
@smarterclayton I would not consider the podinfoserver new API; it's private IPC between the openshift-sdn CNI plugin, and the openshift-sdn node plugin.  It is not intended to be public in any way, and we would reserve the right to change it at any time.

The options:

* root-only private unix domain socket: advantages include a very small amount of well-understood code doing very specific things, which closely follows other examples like Docker's API server and client pattern.  Authentication and security are protected by simple, common Unix mechanisms like filesystem permissions, and we could additionally protect it through peer credentials if we wanted to.  The private API exposed by this socket is small and easily extendable if/when the SDN plugin needs to extend it, but we do not expect or support any other consumers of this API than openshift-sdn.
* talk to api-server: this would require a larger amount of code in the CNI plugin to create Kube and OpenShift clients and find/read the Pod, VNID, HostSubnet, and other objects the CNI plugin requires.  The openshift-sdn node process already has this information because it must track it for other reasons.  Furthermore, this would require passing authentication information into the CNI plugin, including any certificates required and the apiserver IP address/hostname.

> One question I was surprised by - why does the CNI interface not pass enough information down to the network plugin to make a decision there?

CNI can only pass information that kubelet has; that will soon be extended to pass the PodSpec via https://github.com/containernetworking/cni/pull/247.  But that doesn't help openshift-sdn specific stuff like the HostSubnet and the VNID, which still need to be read out-of-band from CNI.